third_party/move/move-vm/runtime/src/native_functions.rs

```diff
@@ -176,19 +188,40 @@ impl<'a, 'b> NativeContext<'a, 'b> {
     }
 
     pub fn type_to_type_layout(&self, ty: &Type) -> PartialVMResult<MoveTypeLayout> {
-        StorageLayoutConverter::new(self.module_storage).type_to_type_layout(ty)
+        Ok(self
+            .type_to_type_layout_with_delayed_field_check(ty)?
```

Comment:
does this now do unnecessary delayed field check and ignore?

Reply:
It is the same as before: `type_to_type_layout` was doing the check and ignoring, I removed this API so that the caller decides if we want to ignore or not, and propagated it up to here. Technically, we can remove this API and let natives handle delayed fields as they see fit, but this adds more changes and this way it is the same logic as before.

Reply:
oh. fair. it's scarier to have the API that can skip a check when it's needed than the other way around, I guess. Maybe whether delayed fields need to be checks can be a parameter but it shouldn't be an overhead or perf issue